packages/atomic/src/components/common/facets/facet-common.tsx

```diff
@@ -253,6 +263,101 @@ export function facetShouldBeInitiallyCollapsed(
   return facetIndex + 1 > collapseFacetsAfter;
 }
 
+interface VisibilitySortedFacets {
+  visibleFacets: BaseFacetElement[];
+  invisibleFacets: BaseFacetElement[];
+}
+export function sortFacetVisibility(
+  facetElements: BaseFacetElement[],
+  facetInfoMap: FacetInfoMap
+): VisibilitySortedFacets {
+  const visibleFacets: BaseFacetElement[] = [];
+  const invisibleFacets: BaseFacetElement[] = [];
+
+  facetElements.forEach((facet) => {
+    if (facetInfoMap[facet.facetId] && facetInfoMap[facet.facetId].isHidden()) {
+      invisibleFacets.push(facet);
+    } else {
+      visibleFacets.push(facet);
+    }
+  });
+
+  return {visibleFacets, invisibleFacets};
+}
+
+export function updateCollapsedState(
+  facets: BaseFacetElement[],
+  collapseFacetsAfter: number
+) {
+  if (collapseFacetsAfter === -1) {
+    return;
+  }
+
+  facets.forEach((facet, index) => {
+    facet.isCollapsed = facetShouldBeInitiallyCollapsed(
+      index,
+      collapseFacetsAfter
+    );
+  });
+}
+export function updateCollapseFacetsAfter(
+  generator: AutomaticFacetGeneratorElement,
+  index: number,
+  collapseFacetsAfter: number
+) {
```

Comment:
The purpose of this function feels ambiguous, I was only able to really understand it by looking at both the code where it gets called and its internal code.

I would prefer removing it altogether, and simply doing this where it was getting called:
```ts
const remainingFacetsToShow = collapseFacetsAfter - visibleFacets.length;
if (generator) {
  generator.collapseFacetsAfter = remainingFacetsToShow;
}
```